### PR TITLE
Always set a "User-Agent" string and let the user override it

### PR DESCRIPTION
**Pull request by ghost**
This change sets a default user agent that will be used in any request. It lets the user override it with a new option to be passed at construction time.

Needs review (of course) and more testing.

### CODE REVIEW

**Comment by ghost** [comment]
By the way, this is my first pull request, so if there are any problems let me know and I'll fix it. Thanks!

**Comment by vankasteelj** [comment]
First, why is a change in the useragent needed? At the moment, the UA is something like 'got', and it reflects the reality, no?

**Comment by ghost** [comment]
I believe the UA should indicate what application is being used to access the content for it to be of any use to anyone. That's why reddit requires each app to set a different one.

My bad for thinking the UA was not being set - I didn't know that `got` sets its own. In light of this, I'd like to change the pull request so that by default you use whatever `got` sets, but the user has the ability to change it.

Would that be ok?

**Comment by janpio** [comment]
Which also makes it a non breaking change as if the users do nothing, nothing changes for them

**Comment by vankasteelj** [comment]
IMO, without looking really deep, it might be better to 

1) set (as you did) the UA in the class constructor, to account for a user-specific UA, something like `useragent: settings.user_agent`, as it will default to undefined if unset

2) for the requests, only the `_call` is really useful if you ask me, the others arent your application, they're the module's internal, and the best way would probably to add after the req={} a `this._settings.useragent && req.headers['User-Agent'] = this._settings.useragent` or something like that. 

Thoughts?

**Comment by vankasteelj** [comment]
and sidenote regarding the pull request itself, i'd rather you split into commits: 

1) the core changes
2) the readme addition
3) the whitespace correction

It adds readability :) 1 commit per change in the code

**Comment by ghost** [comment]
No thoughts other than you gave me a clear path for improvement. I'll get it done today.

**Comment by janpio** [comment]
(I would disagree with only setting the UA in _call. All requests are triggered by the app using trakt.tv as a library. As an API provider I would always want to know (and be able to identify, monitor, block, cache, redirect, fix, manipulate... to) all requests coming from a specific app.)

**Comment by vankasteelj** [comment]
It would be worth asking justin about this, but ok, let's set in on all got calls then. 

edit: there is no useragent specs on the docs nor the g+ community, I asked [here](https://plus.google.com/u/0/+JeanvanKasteel/posts/djekUjA2qn8).

**Comment by vankasteelj** [comment]
Maybe (and I say maybe) it would be worth also to check what UA 'got' sets by default, and if sending a header: {'User-Agent':undefined} removes the default UA or not. Because if it doesnt, there even isnt any need for the '&&' if-like trick i wrote above.

edit: so I checked, and the if trick is needed. Got uses Object.assign() which doesnt overwrite an 'undefined' value.

**Comment by ghost** [comment]
Here's my first attempt at a fix. If we end up with several commits (once we're done working on this) I can close the PR and make a new one with only one commit (if you want to keep the history clean).

File: README.md [hunk 1]
```diff
@@ -24,11 +24,14 @@ let options = {
   client_secret: <the_client_secret>,
   redirect_uri: null,   // defaults to 'urn:ietf:wg:oauth:2.0:oob'
   api_url: null         // defaults to 'https://api.trakt.tv'
+  user_agent: null      // defaults to something like 'Node.js/8.0.0 trakt.tv/v5.1.0'
```
Comment (vankasteelj) [inline README.md:27]: defaults to GOT (http client) user agent

File: README.md [hunk 2]
```diff
@@ -24,11 +24,14 @@ let options = {
...
 };
 const trakt = new Trakt(options);
 ```
 Add `debug: true` to the `options` object to get debug logs of the requests executed in your console.
 
+Version numbers in the default user agent string are determined at runtime.
```
Comment (vankasteelj) [inline README.md:33]: that comment is not needed as of the comment above

File: trakt.js [hunk 3]
```diff
@@ -10,6 +10,8 @@ const crypto = require('crypto');
 const methods = require('./methods.json');
 const sanitizer = require('sanitizer').sanitize;
 
+
+
```
Comment (vankasteelj) [inline trakt.js:14]: no.

File: trakt.js [hunk 4]
```diff
@@ -126,7 +132,7 @@ module.exports = class Trakt {
     // Parse url before api call
     _parse(method, params) {
         if (!params) params = {};
-        
+
```
Comment (vankasteelj) [inline trakt.js:135]: this should be in its own commit, labeled "remove whitespace" or similar

File: package.json [hunk 5]
```diff
@@ -2,7 +2,12 @@
     "name": "trakt.tv",
     "version": "5.1.0",
     "description": "Interact with Trakt.tv API in NodeJS",
-    "keywords": ["trakt", "api", "wrapper", "node"],
+    "keywords": [
+        "trakt",
+        "api",
+        "wrapper",
+        "node"
+    ],
```
Comment (vankasteelj) [inline package.json:10]: there's no need for this either

**Review by vankasteelj — CHANGES_REQUESTED** [review]
looks good to me (with the remarks below)

**Comment by ghost** [comment]
Yeah thank you, I'll have more time this afternoon and make it properly. This has been very instructive :)

**Comment by ghost** [comment]
Btw I think the tags got messed up by npm for whatever reason. I didn't touch them. Strange

**Comment by vankasteelj** [comment]
auto-beautify or auto-lint in your text editor, probably
